### ISSUE TITLE
Minimap: add a `alwaysShowViewport` setting

### ISSUE THREAD

**Issue by SteveVanOpstal** [Microsoft/vscode#21784]
- VSCode Version: 1.10.1
- OS Version: Windows 10

Currently the user is required to hover over the minimap to see the viewport.
A setting which keeps the viewport visible at all time would greatly improve my minimap experience. Something like:

```JSON
"editor.minimap.alwaysShowViewport": true
```

**Comment by jsdnxx** [comment]
related https://github.com/Microsoft/vscode/issues/21810

**Comment by Tekbr** [comment]
@alexandrudima  I added this gif to illustrate the two situations in #21823 

Allow a rectangular shadow to indicate where the minimap is, without having to move the mouse. Like Atom.

![](https://cloud.githubusercontent.com/assets/19866231/23524327/e274029c-ff68-11e6-9978-af344fa1c59e.gif)


And when it slides horizontally, another shadow will be shown to indicate that I'm going to the side. Like Atom.

![](https://cloud.githubusercontent.com/assets/19866231/23524366/0f92c146-ff69-11e6-8475-dcd95c305278.gif)

**Comment by mdimai666** [comment]
Change file
C:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\electron-browser\workbench.main.css

.minimap-slider{opacity:0 --> .minimap-slider{opacity:1

for highlight always

**Comment by deltaidea** [comment]
Is anyone working on this?
If not, I really want this feature and would like to try to implement this.

**Comment by rebornix** [comment]
@deltaidea we didn't start on this yet. You can go ahead :)

**Comment by alexandrudima** [comment]
`"editor.minimap.showSlider": "always"`

**Comment by Tekbr** [comment]
@alexandrudima How to activate horizontally? The same is not working for me, as in Atom. I put a sample gif in this [comment](https://github.com/Microsoft/vscode/issues/21784#issuecomment-284915376).

**Comment by alexandrudima** [comment]
@Tekbr I've extracted the horizontal scroll indicator in the minimap slider to #28498

**Comment by Tekbr** [comment]
-- Sorry for the English, I used Google Translator --

All right, @alexandrudima . You are doing a job, but I believe that the suggestion I have made has significant votes. 👍 


**Note:** I want to make it clear that horizantal does not need to be done right away.

--------------------------------------------------------------------------------------------------------------
Sorry for the outburst. 🤔 😊 

@alexandrudima 
A new case would be to row again ⛵️  to win the votes. (Since the comment already has votes.) Why not leave the case open and say when to check "Check only the first comment (or situation 'X')." You always create a new case for testing yourself, just reference the first comment or say what it is to verify.


You may argue, "It's better to track." But the first criterion for you in the vast majority is always number of votes for a new resource, and it would be difficult, in certain moments to get the votes again.


I had created another question before, you marked it as duplicate, so you think it will come all at once. Well many question in some quetão there is comment similar to this .: "We will expose the ideas in a single case - because it is better to track."

It was much better for you to have changed the title of the first question and to inform me thus "Here only the horizontal resource will be traced."

Then you say that it is for us to look up the questions before creating a new one (ps: yes, not many suggestive names in the questions until you change them - or the difference of language in my case), but when we suggest something Get votes on this question) you ask to create a new one (or in the case we already created it and it was marked as duplicate and we have to create it again.)


You better review the way you deal with these situations, in my opinion.

**NOTE:** This does not include when there is OS difference.

One more excuse for the outburst. 😊 😊

I may even be denied the comment or be hated, but I had to write it this.

**Comment by magic890** [comment]
@alexandrudima ok, close this one but you also close the other one. So no open issue for this feature. When it will be implemented?

**Comment by deltaidea** [comment]
@magic890 It's already implemented. It's currently available on the [Insiders channel](https://code.visualstudio.com/insiders). It'll get into the stable release in a couple weeks I think.

**Comment by alexandrudima** [comment]
@Tekbr I just implemented the horizontal scrolling in #28498 . 

Please consider contributing PRs if you feel very passionate about a certain feature request.

**Comment by Tekbr** [comment]
-- Sorry for the English, I used Google Translator --

Yes, @alexandrudima , but I would be happy to contribute. But I have better enough my English, being that is something main here. In addition to having a longer time to study TypeScript.

Anyway, thank you very much for implementing the feature. ❤️ ❤️ 👏 👏 👏 

But my argument is valid since it was previously marked as a duplicate, and since you ask for a single issue to be tracked, to make it easier, it will soon be interpreted that everything will be done in a single question. That's why I was a little upset.

But everything was solved. Thank you one more time. ❤️ ❤️ 👏 👏 👏
